ms_active_directory/environment/ldap/ldap_format_utils.py

```diff
@@ -368,3 +368,11 @@ def validate_and_normalize_logon_name(name: str, supports_legacy_behavior: bool)
             raise InvalidDomainParameterException('Common names may not contain any of the following characters: '
                                                   '{}'.format(', '.join(AD_USERNAME_RESTRICTED_CHARS)))
     return name
+
+
+def construct_primary_group_sid(object_sid: str, primary_group_id: int) -> str:
+    """ Construct the objectSid for the primary group of an object given the objectSid of the object in question
+    and the primaryGroupID for the object.
+    """
+    base_sid = object_sid[:object_sid.rfind('-')]
+    return '{base}-{rid}'.format(base=base_sid, rid=primary_group_id)
```

Comment:
links to the appropriate microsoft doc would be useful here

Reply:
Done.